In [1]:
import requests
from bs4 import BeautifulSoup
import pandas as pd
from scrape_wiki import scrape_wiki_table
from get_gender_data import get_founder_gender
from filter_df import filter_df

In [2]:
url = "https://en.wikipedia.org/wiki/List_of_unicorn_startup_companies"
idx = 2  # Changed to target the main list of unicorn companies

currentUnicorns_df = scrape_wiki_table(url, idx, "current_unicorns.csv")
currentUnicorns_df.to_csv("current_unicorns.csv")

pastUnicorns_df = scrape_wiki_table(url, idx+1, "past_unicorns.csv")
pastUnicorns_df.to_csv("past_unicorns.csv")

Found 4 wikitable(s)
Saved 616 rows to current_unicorns.csv
Found 4 wikitable(s)
Saved 206 rows to past_unicorns.csv


In [3]:
currentUnicorns_df = pd.read_csv("current_unicorns.csv", index_col=0)
pastUnicorns_df = pd.read_csv("past_unicorns.csv", index_col=0)



In [4]:
display(currentUnicorns_df)
display(pastUnicorns_df)

,Company,Valuation(US$ billions),Valuation date,Industry,Country/countries,Founder(s)
0,SpaceX,1250,February 2026(2026-02)[20],"Aerospace,Artificial Intelligence",United States,Elon Musk
1,OpenAI,852,March 2026(2026-03)[21],Artificial intelligence,United States,"Sam Altman,Elon Musk,Greg Brockman,Ilya Sutskever"
2,Anthropic,380,February 2026(2026-02)[22],Artificial Intelligence,United States,"Dario Amodei,Daniela Amodei,Jared Kaplan, Jack..."
3,ByteDance,330,September 2025[23],Internet,China,"Zhang Yiming, Liang Rubo"
4,Stripe,159,February 2026(2026-02)[24],Financial services,United States and Ireland,PatrickandJohn Collison
...,...,...,...,...,...,...
611,HMD Global,1+,August 2020[569],Mobile Devices,Finland,Jean-Francois Baril
612,IQM,1,July 2022[570],Quantum Computing,Finland,"Jan Goetz, Mikko Möttönen, Kuan Yen Tan, Juha ..."
613,Hostaway,1,December 2024[571],Vacation rental software platform,Finland,"Marcus Räder, Saber Kordestanchi, Mikko Nurminen"
614,Papara,1,July 2023[572],Fintech,Turkey,Ahmed Faruk Karslı


,Company,Last valuation(US$billions),Valuation date,Exit date,Exit reason,Exit valuation(US$billions),Country,Founders,col_8
0,Uber,72,August 2018[575],May 2019[576],IPO,82.4,United States,"Travis Kalanick,Garett Camp",NaN
1,DiDi,62,July 2019[577],June 2021[578],IPO,73,China,Cheng Wei,NaN
2,Facebook,50,January 2011,May 2012[579],IPO,104,United States,"Mark Zuckerberg,Eduardo Saverin, Andrew McColl...",NaN
3,Xiaomi,45,April 2015,July 2018[580],IPO,70,China,Lei Jun,NaN
4,Alibaba,42,June 2016,September 2014[581],IPO,238,China,Jack Ma,NaN
...,...,...,...,...,...,...,...,...,...
201,Zimi,1+,February 2015[184],March 2021[834],Acquired,0.4,China,NaN,NaN
202,QingCloud,1+,June 2017[184],March 2021[835],IPO,0.46,China,NaN,NaN
203,Novogene,1+,November 2016[184],April 2021,IPO,NaN,China,NaN,NaN
204,MissFresh,1+,December 2017[184],June 2021[836],IPO,2.5,China,NaN,NaN


# gender generation

In [5]:
print(currentUnicorns_df.head())
print(pastUnicorns_df.head())

     Company Valuation(US$ billions)              Valuation date  \
0     SpaceX                    1250  February 2026(2026-02)[20]   
1     OpenAI                     852     March 2026(2026-03)[21]   
2  Anthropic                     380  February 2026(2026-02)[22]   
3  ByteDance                     330          September 2025[23]   
4     Stripe                     159  February 2026(2026-02)[24]   

                            Industry          Country/countries  \
0  Aerospace,Artificial Intelligence              United States   
1            Artificial intelligence              United States   
2            Artificial Intelligence              United States   
3                           Internet                      China   
4                 Financial services  United States and Ireland   

                                          Founder(s)  
0                                          Elon Musk  
1  Sam Altman,Elon Musk,Greg Brockman,Ilya Sutskever  
2  Dario Amodei,Daniela

In [8]:
currentUnicorns_df['Founder_Genders'] = filter_df(currentUnicorns_df, "Founder(s)", get_founder_gender)

# Create a new DataFrame with Company and Founder Genders
df_companies_founders_gender = currentUnicorns_df[['Company', 'Founder_Genders', 'Industry']].copy()

df_companies_founders_gender['Founder_Genders'] = df_companies_founders_gender['Founder_Genders'].apply(
    lambda lst: [("male" if v == "mostly_male" else v) for v in lst] if isinstance(lst, (list, tuple)) else lst)

df_companies_founders_gender['Founder_Genders'] = df_companies_founders_gender['Founder_Genders'].apply(
    lambda lst: [("male" if v == "mostly_female" else v) for v in lst] if isinstance(lst, (list, tuple)) else lst)



male
mostly_male
male
male
male
male
female
male
male
male
andy
andy
unknown
male
unknown
male
male
mostly_male
female
male
mostly_male
male
male
male
male
male
male
male
male
male
male
unknown
male
male
male
male
unknown
unknown
male
unknown
male
male
male
female
male
male
male
mostly_male
mostly_male
male
male
unknown
male
male
male
male
male
male
mostly_male
mostly_male
male
male
male
male
male
male
male
male
mostly_male
female
unknown
male
unknown
male
mostly_male
male
male
mostly_male
mostly_male
male
male
male
male
mostly_male
male
male
male
unknown
female
male
female
male
mostly_male
unknown
male
male
male
male
mostly_male
male
male
male
male
male
male
male
male
male
male
male
andy
unknown
male
male
unknown
male
male
male
unknown
male
andy
male
male
male
female
male
unknown
male
male
male
male
male
male
male
male
male
andy
unknown
mostly_male
female
male
male
male
unknown
unknown
male
unknown
male
unknown
male
male
male
male
mostly_male
male
male
unknown
male
male
female
male
ma

# Tabellen-Daten 1: Gender Pro Industrie

In [13]:
# start from currentUnicorns_df which already has Company, Founder_Genders (lists or values), Industry
df = currentUnicorns_df[['Company', 'Founder_Genders', 'Industry']].copy()

# normalize Founder_Genders list items to lowercase and map "mostly_*" to male/female as you did
def normalize_founders(g):
    if isinstance(g, (list, tuple)):
        lst = [str(x).strip().lower() for x in g]
    elif pd.isna(g) or g == '':
        return ['unknown']
    else:
        lst = [str(g).strip().lower()]
    # map variants
    mapped = []
    for v in lst:
        if v == 'mostly_male' or v == 'mostly_female':
            mapped.append('male' if v == 'mostly_male' else 'female')
        else:
            mapped.append(v)
    return mapped

df['Founder_Genders_list'] = df['Founder_Genders'].apply(normalize_founders)

# normalize Industry into a list without using re (replace common separators with '|', then split)
df['Industry'] = df['Industry'].fillna('')
df['Industry_norm'] = (df['Industry']
                       .str.replace(';', '|', regex=False)
                       .str.replace(',', '|', regex=False)
                       .str.replace('/', '|', regex=False)
                       .str.replace('&', '|', regex=False)
                       .str.replace(' and ', '|', regex=False)
                       .str.replace(' & ', '|', regex=False)
                      )
df['Industry_list'] = df['Industry_norm'].apply(lambda s: [x.strip().lower() for x in s.split('|') if x.strip()])

# explode industry and founder-gender lists so each row is one (company, sector, gender)
df_exp = df.explode('Industry_list').reset_index(drop=True)
df_exp = df_exp[df_exp['Industry_list'] != '']  # drop empty sectors
df_exp = df_exp.explode('Founder_Genders_list').reset_index(drop=True)

# canonicalize gender variants and fill unknowns
gender_map = {'m': 'male', 'f': 'female', 'man': 'male', 'woman': 'female'}
df_exp['Founder_Genders_list'] = df_exp['Founder_Genders_list'].replace(gender_map)
df_exp['Founder_Genders_list'] = df_exp['Founder_Genders_list'].fillna('unknown')
df_exp['Founder_Genders_list'] = df_exp['Founder_Genders_list'].astype(str)

# Now aggregate counts per sector
counts = (df_exp
          .groupby([df_exp['Industry_list'].str.title(), 'Founder_Genders_list'])
          .size()
          .unstack(fill_value=0))

# Ensure columns for male, female, unknown exist
for col in ['male', 'female', 'unknown']:
    if col not in counts.columns:
        counts[col] = 0

# Create final DataFrame with reset index and desired column order
sectors_gender_counts = counts.reset_index().rename(columns={'index': 'Sector'})
sectors_gender_counts = sectors_gender_counts.rename(columns={'Industry_list': 'Sector'}) if 'Industry_list' in sectors_gender_counts.columns else sectors_gender_counts
sectors_gender_counts = sectors_gender_counts[['Industry_list' if 'Industry_list' in sectors_gender_counts.columns else counts.index.name, 'male', 'female', 'unknown']] if False else None

# simpler: rebuild final df explicitly
sectors_gender_counts = counts.reset_index().rename(columns={'Industry_list': 'Sector'})
if 'Sector' not in sectors_gender_counts.columns:
    # fallback when reset produced different name
    sectors_gender_counts = counts.reset_index().rename(columns={counts.index.name: 'Sector'})

sectors_gender_counts = sectors_gender_counts[['Sector', 'male', 'female', 'unknown']]

sectors_gender_counts.to_csv('sectors_gender_counts.csv', index=False)

print(sectors_gender_counts)


Founder_Genders_list                         Sector  male  female  unknown
0                                       3D Printing     0       0        1
1                               Accounting Software     0       0        1
2                            Advertising Technology     0       0        2
3                                         Aerospace     1       0        1
4                                       Agriculture     0       0        1
..                                              ...   ...     ...      ...
167                                 Web Development     0       0        1
168                                            Web3     0       0        1
169                                          Webops     0       0        1
170                            Workforce Management     1       0        2
171                   Workforce Management Software     1       0        1

[172 rows x 4 columns]


In [14]:
male_count = 0
female_count = 0
unknown_count = 0

for cell in df_companies_founders_gender['Founder_Genders']:
    if isinstance(cell, (list, tuple)):
        for v in cell:
            if v == 'male':
                male_count += 1
            elif v == 'female':
                female_count += 1
            elif v == 'unknown':
                unknown_count += 1
    else:
        if cell == 'male':
            male_count += 1
        elif cell == 'female':
            female_count += 1
        elif cell == 'unknown':
            unknown_count += 1

print("male:", male_count)
print("female:", female_count)
print("unknown:", unknown_count)


male: 305
female: 23
unknown: 88


In [12]:
unknown_names = []

for i, cell in df_companies_founders_gender['Founder_Genders'].items():
    if isinstance(cell, (list, tuple)):
        for name in cell:
            if name == 'unknown':
                unknown_names.append((i, name))
    else:
        if cell == 'unknown':
            unknown_names.append((i, cell))

# Print rows (index) that have unknowns and the count per row
for idx, _ in unknown_names:
    print(f"Row {idx} contains 'unknown'")

# If you want the total and unique indices:
print("Total unknowns:", len(unknown_names))
print("Rows with unknowns:", sorted({idx for idx, _ in unknown_names}))


Row 3 contains 'unknown'
Row 3 contains 'unknown'
Row 4 contains 'unknown'
Row 6 contains 'unknown'
Row 15 contains 'unknown'
Row 16 contains 'unknown'
Row 17 contains 'unknown'
Row 18 contains 'unknown'
Row 23 contains 'unknown'
Row 34 contains 'unknown'
Row 35 contains 'unknown'
Row 43 contains 'unknown'
Row 48 contains 'unknown'
Row 56 contains 'unknown'
Row 57 contains 'unknown'
Row 58 contains 'unknown'
Row 60 contains 'unknown'
Row 63 contains 'unknown'
Row 71 contains 'unknown'
Row 82 contains 'unknown'
Row 82 contains 'unknown'
Row 88 contains 'unknown'
Row 88 contains 'unknown'
Row 90 contains 'unknown'
Row 91 contains 'unknown'
Row 101 contains 'unknown'
Row 127 contains 'unknown'
Row 127 contains 'unknown'
Row 127 contains 'unknown'
Row 127 contains 'unknown'
Row 132 contains 'unknown'
Row 141 contains 'unknown'
Row 141 contains 'unknown'
Row 152 contains 'unknown'
Row 152 contains 'unknown'
Row 166 contains 'unknown'
Row 166 contains 'unknown'
Row 175 contains 'unknown'
Row

# Tabelle-Daten 2: Land + Geschlecht

In [17]:
import pandas as pd

# start from currentUnicorns_df which already has Company, Founder_Genders, and "Country/countries"
df = currentUnicorns_df[['Company', 'Founder_Genders', 'Country/countries']].copy()

# normalize Founder_Genders list items to lowercase and map "mostly_*" to male/female
def normalize_founders(g):
    if isinstance(g, (list, tuple)):
        lst = [str(x).strip().lower() for x in g]
    elif pd.isna(g) or g == '':
        return ['unknown']
    else:
        lst = [str(g).strip().lower()]
    mapped = []
    for v in lst:
        if v == 'mostly_male':
            mapped.append('male')
        elif v == 'mostly_female':
            mapped.append('female')
        else:
            mapped.append(v)
    return mapped

df['Founder_Genders_list'] = df['Founder_Genders'].apply(normalize_founders)

# normalize "Country/countries" into list (no regex)
col = 'Country/countries'
df[col] = df[col].fillna('')
df[col + '_norm'] = (df[col]
                     .str.replace(';', '|', regex=False)
                     .str.replace(',', '|', regex=False)
                     .str.replace('/', '|', regex=False)
                     .str.replace('&', '|', regex=False)
                     .str.replace(' and ', '|', regex=False)
                     .str.replace(' & ', '|', regex=False)
                    )
df[col + '_list'] = df[col + '_norm'].apply(lambda s: [x.strip().lower() for x in s.split('|') if x.strip()])

# explode so each row is one (company, country, gender)
df_exp = df.explode(col + '_list').reset_index(drop=True)
df_exp = df_exp[df_exp[col + '_list'] != '']  # drop empty countries
df_exp = df_exp.explode('Founder_Genders_list').reset_index(drop=True)

# canonicalize genders and fill unknown
gender_map = {'m': 'male', 'f': 'female', 'man': 'male', 'woman': 'female'}
df_exp['Founder_Genders_list'] = df_exp['Founder_Genders_list'].replace(gender_map)
df_exp['Founder_Genders_list'] = df_exp['Founder_Genders_list'].fillna('unknown')
df_exp['Founder_Genders_list'] = df_exp['Founder_Genders_list'].astype(str)

# aggregate counts per country
counts = (df_exp
          .groupby([df_exp[col + '_list'].str.title(), 'Founder_Genders_list'])
          .size()
          .unstack(fill_value=0))

# ensure columns exist
for gender in ['male', 'female', 'unknown']:
    if gender not in counts.columns:
        counts[gender] = 0

# Build final DataFrame with column name matching "Country/countries"
countries_gender_counts = counts.reset_index().rename(columns={col + '_list': 'Country/countries'})
countries_gender_counts = countries_gender_counts.rename(columns={countries_gender_counts.columns[0]: 'Country/countries'})
countries_gender_counts = countries_gender_counts[['Country/countries', 'male', 'female', 'unknown']]

# save to CSV
countries_gender_counts.to_csv('countries_gender_counts.csv', index=False)


# Tabellendaten 4: Exit Valuation

In [22]:
pastUnicorns_df['Founder_Genders'] = filter_df(pastUnicorns_df, "Founders", get_founder_gender)

# Create a new DataFrame with Company and Founder Genders
df_companies_founders_gender = pastUnicorns_df[['Company', 'Founder_Genders', 'Exit valuation(US$billions)']].copy()

df_companies_founders_gender['Founder_Genders'] = df_companies_founders_gender['Founder_Genders'].apply(
    lambda lst: [("male" if v == "mostly_male" else v) for v in lst] if isinstance(lst, (list, tuple)) else lst)

df_companies_founders_gender['Founder_Genders'] = df_companies_founders_gender['Founder_Genders'].apply(
    lambda lst: [("male" if v == "mostly_female" else v) for v in lst] if isinstance(lst, (list, tuple)) else lst)


male
male
andy
male
male
male
mostly_male
male
andy
male
male
male
male
male
male
male
female
andy
andy
unknown
male
male
female
male
male
male
male
male
male
male
male
unknown
mostly_male
mostly_male
male
male
male
male
male
mostly_male
male
male
male
unknown
male
male
male
unknown
mostly_male
male
male
male
male
unknown
male
male
male
male
male
male
male
unknown
male
male
male
female
male
male
male
male
male
male
male
unknown
male
male
male
male
male
male
unknown
mostly_male
male
male
male
male
male
male
male
male
unknown
unknown
male
male
male
male
male
male
male
female
mostly_male
male
unknown
andy
unknown
unknown
unknown
male
unknown
male
male
male
male
mostly_male
unknown
andy
female
male
male
male
male
male
mostly_male
mostly_male
male
unknown
male
unknown
male
male
unknown
male
unknown
male
male
male
male
male
unknown
unknown
unknown
male
male
unknown
unknown
unknown
male
mostly_male
female
male
male
male
male
male
male
male
male
male
male
male
male
male
female


In [23]:
import pandas as pd

# start from currentUnicorns_df which has Company, Founder_Genders, and "Exit valuation(US$billions)"
df = pastUnicorns_df[['Company', 'Founder_Genders', 'Exit valuation(US$billions)']].copy()

# normalize Founder_Genders list items to lowercase and map "mostly_*" to male/female
def normalize_founders(g):
    if isinstance(g, (list, tuple)):
        lst = [str(x).strip().lower() for x in g]
    elif pd.isna(g) or g == '':
        return ['unknown']
    else:
        lst = [str(g).strip().lower()]
    mapped = []
    for v in lst:
        if v == 'mostly_male':
            mapped.append('male')
        elif v == 'mostly_female':
            mapped.append('female')
        else:
            mapped.append(v)
    return mapped

df['Founder_Genders_list'] = df['Founder_Genders'].apply(normalize_founders)

# normalize "Exit valuation(US$billions)" into list (no regex)
col = 'Exit valuation(US$billions)'
df[col] = df[col].fillna('')
df[col + '_norm'] = (df[col]
                     .astype(str)
                     .str.replace(';', '|', regex=False)
                     .str.replace(',', '|', regex=False)
                     .str.replace('/', '|', regex=False)
                     .str.replace('&', '|', regex=False)
                     .str.replace(' and ', '|', regex=False)
                     .str.replace(' & ', '|', regex=False)
                    )
df[col + '_list'] = df[col + '_norm'].apply(lambda s: [x.strip() for x in s.split('|') if x.strip()])

# explode so each row is one (company, exit_val, gender)
df_exp = df.explode(col + '_list').reset_index(drop=True)
df_exp = df_exp[df_exp[col + '_list'] != '']  # drop empty entries
df_exp = df_exp.explode('Founder_Genders_list').reset_index(drop=True)

# canonicalize genders and fill unknown
gender_map = {'m': 'male', 'f': 'female', 'man': 'male', 'woman': 'female'}
df_exp['Founder_Genders_list'] = df_exp['Founder_Genders_list'].replace(gender_map)
df_exp['Founder_Genders_list'] = df_exp['Founder_Genders_list'].fillna('unknown')
df_exp['Founder_Genders_list'] = df_exp['Founder_Genders_list'].astype(str)

# aggregate counts per exit valuation
counts = (df_exp
          .groupby([df_exp[col + '_list'], 'Founder_Genders_list'])
          .size()
          .unstack(fill_value=0))

# ensure columns exist
for gender in ['male', 'female', 'unknown']:
    if gender not in counts.columns:
        counts[gender] = 0

# Build final DataFrame with original column name
exitval_gender_counts = counts.reset_index().rename(columns={col + '_list': col})
# ensure column order
exitval_gender_counts = exitval_gender_counts[[col, 'male', 'female', 'unknown']]

# save to CSV
exitval_gender_counts.to_csv('exitval_gender_counts.csv', index=False)


In [24]:
import pandas as pd

# start from currentUnicorns_df which has Company, Founder_Genders, and "Exit valuation(US$billions)"
df = pastUnicorns_df[['Company', 'Founder_Genders', 'Exit valuation(US$billions)']].copy()

# normalize Founder_Genders list items to lowercase and map "mostly_*" to male/female
def normalize_founders(g):
    if isinstance(g, (list, tuple)):
        lst = [str(x).strip().lower() for x in g]
    elif pd.isna(g) or g == '':
        return ['unknown']
    else:
        lst = [str(g).strip().lower()]
    mapped = []
    for v in lst:
        if v == 'mostly_male':
            mapped.append('male')
        elif v == 'mostly_female':
            mapped.append('female')
        else:
            mapped.append(v)
    return mapped

df['Founder_Genders_list'] = df['Founder_Genders'].apply(normalize_founders)

# normalize "Exit valuation(US$billions)" into list (no regex)
col = 'Exit valuation(US$billions)'
df[col] = df[col].fillna('')
df[col + '_norm'] = (df[col]
                     .astype(str)
                     .str.replace(';', '|', regex=False)
                     .str.replace(',', '|', regex=False)
                     .str.replace('/', '|', regex=False)
                     .str.replace('&', '|', regex=False)
                     .str.replace(' and ', '|', regex=False)
                     .str.replace(' & ', '|', regex=False)
                    )
df[col + '_list'] = df[col + '_norm'].apply(lambda s: [x.strip() for x in s.split('|') if x.strip()])

# explode so each row is one (company, exit_val, gender)
df_exp = df.explode(col + '_list').reset_index(drop=True)
df_exp = df_exp[df_exp[col + '_list'] != '']  # drop empty entries
df_exp = df_exp.explode('Founder_Genders_list').reset_index(drop=True)

# canonicalize genders and fill unknown
gender_map = {'m': 'male', 'f': 'female', 'man': 'male', 'woman': 'female'}
df_exp['Founder_Genders_list'] = df_exp['Founder_Genders_list'].replace(gender_map)
df_exp['Founder_Genders_list'] = df_exp['Founder_Genders_list'].fillna('unknown')
df_exp['Founder_Genders_list'] = df_exp['Founder_Genders_list'].astype(str)

# aggregate counts and collect company names per exit valuation
agg_counts = (df_exp
              .groupby([col + '_list', 'Founder_Genders_list'])
              .size()
              .unstack(fill_value=0))

# ensure columns exist
for gender in ['male', 'female', 'unknown']:
    if gender not in agg_counts.columns:
        agg_counts[gender] = 0

# collect unique company names per exit valuation (semicolon-separated)
companies = (df_exp.groupby(col + '_list')['Company']
             .apply(lambda s: '; '.join(sorted(pd.unique(s.dropna().astype(str)))))
            )

# merge counts with companies Series
result = agg_counts.reset_index().rename(columns={col + '_list': col})
result = result.merge(companies.reset_index().rename(columns={col + '_list': col}), on=col)

# ensure column order
result = result[[col, 'male', 'female', 'unknown', 'Company']].rename(columns={'Company': 'companies'})

# save to CSV
result.to_csv('exitval_gender_counts_with_companies.csv', index=False)


# Planung

Stats:
male: 305
female: 23
unknown: 88

Tabelle 1: Industry + (male, female, unknown)


Tabelle 2: Land + (male, female, unknown)


Tabelle 3: Top 5 Industry in Land + (male, female, unknown)


Tabelle 4: EXIT Wert pro (male, female, unkown)


Visualisierungen